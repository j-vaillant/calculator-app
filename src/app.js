import React, { Component } from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux'


import Key from './components/Key';
import Label from './components/Label';

import { calculatorActions } from './state/actions';

const Container = styled.div`
  position: absolute;
  left: 50%;
  top: 50%;

  display: flex;
  flex-direction: column;

  transform: translate(-50%, -50%);

  width: 800px;
  height: 600px;
`;

const Calc = styled.div`
  display: flex;
  flex: 1 0 500px;
  background-color: #D3D0D0;
`;

const Header = styled.div`
  flex: 1 0 100px; 
  background-color: transparent;
  display: flex;
  flex-direction: column;
  border: 1px solid rgba(0,0,0,0.1);
  box-sizing: border-box;
  padding: 15px;
`;

const Column = styled.div`
  display: flex;
  
  flex-direction: column
  flex: 1 0 auto;
`;


const Calculator = class Calculator extends Component {
  onDigitTyped = (value) => () => {
    this.props.dispatch(calculatorActions.typeNumber(value))
  }
  render() {
    return (
        <Container>
        <Header>
        <Label text={this.props.input} />
        <Label text='10' align='right'/>
        </Header>
        <Calc>
            <Column>
            <Key primary>C</Key>
            <Key onClick={this.onDigitTyped(7)}>7</Key>
            <Key onClick={this.onDigitTyped(4)}>4</Key>
            <Key onClick={this.onDigitTyped(1)}>1</Key>
            <Key onClick={this.onDigitTyped(0)}>0</Key>
            </Column>
            <Column>
            <Key primary>&times;</Key>
            <Key onClick={this.onDigitTyped(8)}>8</Key>
            <Key onClick={this.onDigitTyped(5)}>5</Key>
            <Key onClick={this.onDigitTyped(2)}>2</Key>
            <Key></Key>
            </Column>
            <Column>
            <Key primary> &divide;</Key>
            <Key onClick={this.onDigitTyped(9)}>9</Key>
            <Key onClick={this.onDigitTyped(6)}>6</Key>
            <Key onClick={this.onDigitTyped(3)}>3</Key>
            <Key big>.</Key>
            </Column>
            <Column>
            <Key></Key>
            <Key primary>+</Key>
            <Key primary>-</Key>
            <Key primary merged bg={'#2ABAE9'}>=</Key>
            </Column>
        </Calc>
        </Container>
    );
  }
};

export default connect(({input}) => ({input}))(Calculator);
