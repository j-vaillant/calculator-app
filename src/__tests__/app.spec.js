import React from 'react';
import { shallow } from 'enzyme';

import { PureCalculator } from '../app';

const getProps = value => ({
  input: '',
  result: '',
  mode: 'user',
  ...value,
});

describe('App', () => {
  it('should render correctly', () => {
    const tree = shallow(<PureCalculator {...getProps()} />);

    expect(tree).toMatchSnapshot();
  });

  it('should call props.switchMode() when user changes', () => {
    const mock = jest.fn();
    const tree = shallow(<PureCalculator {...getProps({ switchMode: mock })} />);

    tree.find('Toggler').simulate('click');

    expect(mock).toBeCalled();
  });
});
