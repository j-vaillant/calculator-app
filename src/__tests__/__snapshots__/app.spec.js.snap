// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<styled.div>
  <styled.div>
    <Label
      align="left"
      text=""
    />
    <Label
      align="right"
      text=""
    />
  </styled.div>
  <styled.div>
    <styled.div>
      <Key
        bg={null}
        big={false}
        disabled={false}
        merged={false}
        onClick={[Function]}
        primary={true}
        text=""
      >
        C
      </Key>
      <Key
        bg={null}
        big={false}
        disabled={false}
        merged={false}
        onClick={[Function]}
        primary={false}
        text=""
      >
        7
      </Key>
      <Key
        bg={null}
        big={false}
        disabled={false}
        merged={false}
        onClick={[Function]}
        primary={false}
        text=""
      >
        4
      </Key>
      <Key
        bg={null}
        big={false}
        disabled={false}
        merged={false}
        onClick={[Function]}
        primary={false}
        text=""
      >
        1
      </Key>
      <Key
        bg={null}
        big={false}
        disabled={false}
        merged={false}
        onClick={[Function]}
        primary={false}
        text=""
      >
        0
      </Key>
    </styled.div>
    <styled.div>
      <Key
        bg={null}
        big={false}
        disabled={false}
        merged={false}
        onClick={[Function]}
        primary={true}
        text=""
      >
        ร
      </Key>
      <Key
        bg={null}
        big={false}
        disabled={false}
        merged={false}
        onClick={[Function]}
        primary={false}
        text=""
      >
        8
      </Key>
      <Key
        bg={null}
        big={false}
        disabled={false}
        merged={false}
        onClick={[Function]}
        primary={false}
        text=""
      >
        5
      </Key>
      <Key
        bg={null}
        big={false}
        disabled={false}
        merged={false}
        onClick={[Function]}
        primary={false}
        text=""
      >
        2
      </Key>
      <Key
        bg={null}
        big={false}
        disabled={true}
        merged={false}
        onClick={null}
        primary={false}
        text=""
      />
    </styled.div>
    <styled.div>
      <Key
        bg={null}
        big={false}
        disabled={false}
        merged={false}
        onClick={[Function]}
        primary={true}
        text=""
      >
         รท
      </Key>
      <Key
        bg={null}
        big={false}
        disabled={false}
        merged={false}
        onClick={[Function]}
        primary={false}
        text=""
      >
        9
      </Key>
      <Key
        bg={null}
        big={false}
        disabled={false}
        merged={false}
        onClick={[Function]}
        primary={false}
        text=""
      >
        6
      </Key>
      <Key
        bg={null}
        big={false}
        disabled={false}
        merged={false}
        onClick={[Function]}
        primary={false}
        text=""
      >
        3
      </Key>
      <Key
        bg={null}
        big={true}
        disabled={false}
        merged={false}
        onClick={[Function]}
        primary={false}
        text=""
      >
        .
      </Key>
    </styled.div>
    <styled.div>
      <Key
        bg={null}
        big={false}
        disabled={true}
        merged={false}
        onClick={null}
        primary={false}
        text=""
      />
      <Key
        bg={null}
        big={false}
        disabled={false}
        merged={false}
        onClick={[Function]}
        primary={true}
        text=""
      >
        +
      </Key>
      <Key
        bg={null}
        big={false}
        disabled={false}
        merged={false}
        onClick={[Function]}
        primary={true}
        text=""
      >
        -
      </Key>
      <Key
        bg="#2ABAE9"
        big={false}
        disabled={false}
        merged={true}
        onClick={[Function]}
        primary={true}
        text=""
      >
        =
      </Key>
    </styled.div>
  </styled.div>
  <Toggler
    mode="user"
    onClick={[Function]}
  />
</styled.div>
`;
